### PR TITLE
You can now navigate between dashboard and log pages

### LINKED ISSUE #1688: Make transaction log reachable from navigation
The log page is in the app, but nothing points to it yet!

This feature should implement the new header icons found in this Figma: https://www.figma.com/file/gmhVq6Pqeu2gH7e3FC2lr0ax/Unlock-Dashboard?node-id=0%3A67

I think there was some question about the icons, @smombartz I assume the 2 new icons map as follows:

- Home icon -> Dashboard
- List icon -> Log page

Is that correct?

### PR DESCRIPTION
**Pull request by cnasc**
# Description

This PR adds 2 buttons into the header. They are only shown when the user is on an app page--right now that's just the dashboard and the log page. As a result, it is now possible for end-users to use the log page without being clairvoyant and guessing the URL.

Homepage: same as before
image

Dashboard: addition of buttons in the header (active button highlighted)
image

Creator log: new buttons in header, active page highlighted
image

# Issues

Fixes #1688 #1805 

# Checklist:

- [x] 1 PR, 1 purpose: my Pull Request applies to a single purpose
  - [ ] This PR only contains configuration changes (package.json, etc.)
  - [x] This PR only contains code changes (if configuration changes are required, do a separate PR first, then re-base)
- [x] My code follows the style guidelines of this project, including naming conventions
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] If my code adds or changes components, I have written corresponding stories with Storybook
- [x] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread (TODO)

### CODE REVIEW

File: unlock-app/src/components/interface/Header.js [hunk 1]
```diff
@@ -54,6 +62,21 @@ export default class Header extends React.PureComponent {
             {title}
           </Title>
         )}
+        <AppButtons>
+          {onAppPage &&
+            appButtons.map(({ Button, page }) => {
+              const isActivePage = page === pathname
+              return (
+                <Button
```
Comment (cellog) [inline unlock-app/src/components/interface/Header.js]: can we abstract this behavior into a PageNavButton or something?

Comment (cnasc) [reply unlock-app/src/components/interface/Header.js]: I don't see why not! Keep it local to `Header` or have it in a separate file?

Comment (cellog) [reply unlock-app/src/components/interface/Header.js]: I'd put it with the other buttons, in its own file, since we will probably add other pages later, this sets up a pattern to follow

Comment (cnasc) [reply unlock-app/src/components/interface/Header.js]: Alright, I'll do that as a separate PR then rebase this one just to keep it from swelling too hugely

Comment (cnasc) [reply unlock-app/src/components/interface/Header.js]: The buttons were kind of mixed in with the work I had going here... Just crammed it all in here at least for the moment, let me know if I should split.

**Comment by cnasc** [comment]
Now that #1917 and #1971 are fixed I think this is ready for another review

File: unlock-app/src/__tests__/components/interface/GlobalErrorConsumer.test.js [hunk 2]
```diff
@@ -1,28 +1,29 @@
...
+import { Provider } from 'react-redux'
 import GlobalErrorConsumer, {
   displayError,
 } from '../../../components/interface/GlobalErrorConsumer'
 import { GlobalErrorContext } from '../../../utils/GlobalErrorProvider'
 import { FATAL_NO_USER_ACCOUNT, FATAL_MISSING_PROVIDER } from '../../../errors'
+import createUnlockStore from '../../../createUnlockStore'
 
-const Provider = GlobalErrorContext.Provider
+const ErrorProvider = GlobalErrorContext.Provider
```
Comment (julien51) [inline unlock-app/src/__tests__/components/interface/GlobalErrorConsumer.test.js]: I am a bit confused as to why these changes need to be in this PR?

Comment (cnasc) [reply unlock-app/src/__tests__/components/interface/GlobalErrorConsumer.test.js]: No problem at all, I will get started on splitting this!
